### PR TITLE
Collapse empty table cells and headers

### PR DESCRIPTION
**Pull request by cskrov**
### Description
In some cases, at least ours, there are situations where columns in tables may be completely empty.
In our case because of asynchronous conditional components.

_In most cases the column should be removed._

#### Before and after
In the example below, the 3rd column, "Etternavn", has been emptied. The header and cells are still there, but have no content.
In the first image, between the "Fornavn" and "Rolle" columns, the gap is visually doubled due to the empty column.
table-empty-before\
table-empty-after
In the second image, the gap is even between all columns with content. The empty "Etternavn" column does not add visually misaligning spacing.

### Component Checklist 📝

- [ ] JSDoc
- [ ] Examples
- [ ] Documentation
- [ ] Storybook
- [ ] Style mappings (`@navikt/core/css/config/_mappings.js`)
- [ ] Component tokens (`@navikt/core/css/tokens.json`)
- [ ] CSS class deprecations (`@navikt/aksel-stylelint/src/deprecations.ts`)
- [ ] Exports (`@navikt/core/react/src/index.ts` and `@navikt/core/react/package.json`)
- [ ] New component? CSS import (`@navikt/core/css/index.css`)
- [ ] Breaking change? Update migration guide. Consider codemod.
- [x] Changeset (Format: `<Component>: <gitmoji?> <Text>.` E.g. "Button: :sparkles: Add feature xyz.")

### CODE REVIEW

**Comment by KenAJoh** [comment]
Thank you for the PR 👏  

As you said this would be the prefered way to handle this
> In most cases the column should be removed.

but there are ofc always some exceptions. How does the application know to remove the "Etternavn"-text from table-header here?

**Comment by cskrov** [comment]
> but there are ofc always some exceptions. How does the application know to remove the "Etternavn"-text from table-header here?

Sorry for the confusion, the example was not meant as an example of our exact use case, just as an example of the change in general.

We have no header for these columns, they contain self describing functionality.

Our most common use case is not wasting space by adding the header "Åpne" above buttons (links) with the text "Åpne" (which might not be there).

If this is too broad to add directly, could hiding this functionality behind a flag, ex. `collapseEmptyCells?: boolean`, be on the table? _Pun intended._

**Comment by KenAJoh** [comment]
Would this be a somewhat realistic example of todays issue?
[Normally looks like this](https://aksel.nav.no/sandbox/preview/index.html?code=N4Igxg9gJgpiBcJgApkEoAEBeAfB4AOgHYYaREDOALhgKIAeAhgLYAOANjNhutnoSVIYATjCoBXYSWTEhQgDwAVRgCNOOWXIXK1MAHQAJGI1jCNgrdtWc9AJQgB3c5ctLr+oyZjCAwjHbsGBSQrDBYBOAQ7BEYAPTOLnJuuobGpn4BQSFhEZDRIDgAYgAfUBT+FETCevKxOjae6f7sCYmkyQ1p3hmBwRCh4ZH5OADKVIzCVLX1Hl2+za0u0+52jotJdSuN3utWKQBC0ACeu3LAUIzjesyMrKjAGEQsMAA0GABmVW-UEzQAvm8AJaYXD4TRtERiSTScEQjAdfT2BwYADWMCOWGAgIwAGoPlU-qc2gi9AARS6MHpZfo5EDCRwRIkQrEYACkGAATBgAPzw-biKhUCAkCiAgBetPoFBuAUZAFHWEQYLV+YLhXh4I9xAE-rC4fDNilyeMekyljMyRTTcBPsI-ssjVaFnq4STjZSFjaIMIblRkErkcaYMgfpM0Gh7YabO7TS7Eg6bEizaQ0ABuONCP4RjMJ-SHKAnF25xZpzR-dOCcjUD7e33cZAXKgwEH8WFVmhHbiN-QAczEhW17AAmsZhOgK3J2xhmPXu3o+1QALLCqgAC14eIAjGg9EKxsJAUQe+g9KwTGNfsgOW8IgAGCKlixkYXVqBdy69sRBk97qgHo8nmeUAXpMV43iA94gI+ciiBIUgYAABgAJOcfx6ChzBoShRx-AhE4YOWmhTo2jDcAA2rCAiWE8zAwJqEQAFKMCiEAqOURBvIuEwouIFARC8Lq2vREEAMybhyACcHIAGwAKwACzSSJ-EuqGVDCRyt4cpuAC0t7yTpHIAByKJuEnwOJFnyXoImyUZABaKlPgClEujRdEYBEBiMOwABu3rsZxEyAk5lhCZ5ICbtJt5KUp8mbiJ8mhVoakaVpHJ6bpkmKLe5myZuFkSXoRkiSJjkgAJzmVWcbnPMJPirge1CAowlAwBxGBcWuLV8RVglVMJm5GeJ8mjZu8kiUZyVyKlEWadpenSXp8mKJp8CyQA7PAkl6JuskSeV1WZkdpBUVo7nCYUohQAeKKBRgAAyxirtNQjhREHLybeRkSd9t5DZBJ1CLNH1abp306WJpmyfA8lbSJ0l6JJZWvS5T5nXIF0RbQvntfdDHCowvVA6Q72RcNMWfUZimA6p4yTGlC23htOmbhtpkFQl22I5uUWHbCaOkAAuhWmiwdC8IMCwHBcPEFZZugupECAfxAA)

[But ends up like this in some edgecases](https://aksel.nav.no/sandbox/preview/index.html?code=N4Igxg9gJgpiBcJgApkEoAEBeAfB4AOgHYYaREDOALhgKIAeAhgLYAOANjNhutnoSVIYATjCoBXYSWTEhQgDwAVRgCNOOWXIXK1MAHQAJGI1jCNgrdtWc9AJQgB3c5ctLr+oyZjCAwjHbsGBSQrDBYBOAQ7BEYAPTOLnJuuobGpn4BQSFhEZDRIDgAYgAfUBT+FETCevKxOjae6f7sCYmkyQ1p3hmBwRCh4ZH5OADKVIzCVLX1Hl2+za0u0+52jotJdSuN3utWKQBC0ACeu3LAUIzjesyMrKjAGEQsMAA0GABmVW-UEzQAvm8AJaYXD4TRtERiSTScEQjAdfT2BwYADWMCOWGAgIwAGoPlU-qc2gi9AARS6MHpZfo5EDCRwRHDLFLk8Y9ImJEmsykLYCfYR-Zk2bns2Fw+EzMkU9l8iDCG5UZBEGDI1kwZA-SZoNCCzYs6ULMXEvU2JEcoRoADcRrkfx1NvhJv0hygJyNQpgiytmj+1sE5GoHzlCu4yAuVBgIP4sIDNCO3HD+gA5mJCuIAgBNYzCdB+uSxjDMUOJvQpqgAWQgRCoAAteHiAIxoPRUCBjYSAohJ9B6VgmMa-ZAAJjeEQADBFvRYyFXA1AE5dk2I1T3W+3O93m32oAPJsPRyAJyAp3JRBIpBgAAYAEnOfz0t+Y99vRz+l7zGF9mgL4cY3AA2rCAiWE8zAwPAGARAAUowKIQCo5REG85YTCi4gUBELxGvyEHjgAzA2Q4AJxDgAbAArAALKReGYUampULhIBDmOQ4NgAtGOlHsUOAAcigNkR8CEcJlF6Hh5G8QAWnR04AkBRqgeBkEgAYjDsAAbnKiHIRMgKyZYOEqQ2pFjjRNGUQ2eGUQZWgMUxLFDkOnEccRihjkJ5ENsJRF6LxeF4TJIBYXJIVnIpzxMT4NYdtQgKMJQMBIRgKG1vFGHBdhVRMQ2vGEZRBUNpReG8bZcj2SpjkcWZnGUYoLHwORADs8DEXoDbkURQVhUI8nTsBWhKUxhSiFAHYojpGAADLGDWZVCEZERDpRY68URq1jrlR49eV4yTA5rHVbx7EEQJ5HwJRLV4aRejEYF819eF06kENKm0BpSWTVBVaMBlO0Ldlxl5WZy28dR230XtjGVYdnFNexDZNQJ3lWa1N0NiZ3Wwo9GAALp+poZ7QvCDAsBwXDxH6droH8xAgH8QA)

**Comment by cskrov** [comment]
Those look like simplified versions of our two scenarios. Although I would not call the second one an "edgecase".

File: .changeset/fifty-plums-promise.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+"@navikt/ds-css": patch
+---
+
+Collapses empty table cells and headers by removing all padding. Makes completely empty columns invisible.
```
Comment (KenAJoh) [inline .changeset/fifty-plums-promise.md]:
```suggestion
Table: Collapses empty table cells and headers by removing padding. Makes completely empty columns invisible.
```

**Review by KenAJoh — APPROVED** [review]

**Comment by KenAJoh** [comment]
We have been a little back and forth now on if this is something we want to "fix", since it is technically not allowed (because of screen-readers etc). We tried finding some alternative solutions, and made this CSS-snippet in the process. It will set each `td` to display:none if each element is empty: https://play.tailwindcss.com/InCCbGzXmj?file=css. Would this work as a solution for your case? We will not implement it inside our Table-component, so will need to be added locally.

**Comment by cskrov** [comment]
Ok, thanks, yeah, that is a better solution. We will add it locally.

I does seem like you solved the accessibility problem though.
It makes me wonder, what prevents you from adding the new and improved version to Aksel?

**Pull request closed without merging**